packages/vite/src/node/plugins/css.ts

```diff
@@ -1050,9 +1065,17 @@ interface PostCSSConfigResult {
 }
 
 async function resolvePostcssConfig(
-  config: ResolvedConfig
+  config: ResolvedConfig,
+  dialect = 'css'
 ): Promise<PostCSSConfigResult | null> {
-  let result = postcssConfigCache.get(config)
+  if (!postcssConfigCache[dialect]) {
+    postcssConfigCache[dialect] = new WeakMap<
+      ResolvedConfig,
+      PostCSSConfigResult | null
+    >()
+  }
```

Comment:
I think we can use something like following to shorten the code

```suggestion
  postcssConfigCache[dialect] ??= new WeakMap<
    ResolvedConfig,
    PostCSSConfigResult | null
  >()
```